src/libraries/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.LegacyAlgorithms.cs

```diff
@@ -0,0 +1,31 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Runtime.InteropServices;
+
+internal static partial class Interop
+{
+    internal static partial class Crypto
+    {
+        private static bool s_loadedLegacy;
+        private static object s_legacyLoadLock = new object();
+
+        [DllImport(Libraries.CryptoNative)]
+        private static extern void CryptoNative_RegisterLegacyAlgorithms();
+
+        internal static void RegisterLegacyAlgorithms()
+        {
+            if (!s_loadedLegacy)
+            {
+                lock (s_legacyLoadLock)
+                {
+                    if (!s_loadedLegacy)
+                    {
+                        CryptoNative_RegisterLegacyAlgorithms();
+                        s_loadedLegacy = true;
```

Comment:
@stephentoub If there's a run-once check that makes you happier, let me know :smile:.

I considered a `Lazy<bool>`, but that felt like a sledgehammer when I needed a thumbtack.

Reply:
s_loadedLegacy should be volatile.  Other than that, this looks fine.

I might also rename RegisterLegacyAlgorithms to EnsureLegacyAlgorithmsRegistered, since each call isn't actually registering.